### PR TITLE
fix: make sure to fall back to old query path when query job is incomplete

### PR DESCRIPTION
**Pull request by stephaniewang526**
When the query job takes more than 10s we must fall back to the old query path!

cc @epavan

### CODE REVIEW

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java [hunk 1]
```diff
@@ -1268,7 +1268,7 @@ private TableResult queryRpc(
 
     long numRows;
     Schema schema;
-    if (results.getSchema() == null && results.getJobComplete()) {
+    if ((results.getSchema() == null && results.getJobComplete()) || !results.getJobComplete()) {
```
Comment (epavan123) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java:1271]: This block reads quite confusing now.

consider flipping it..

if (results.getJobComplete() && results.getSchema() != null) {
  schema = Schema.fromPb(results.getSchema());
  // Your logic below for numRows, numDml...
} else {
  JobId jobId = ....
....
}

File: google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java [hunk 2]
```diff
@@ -1813,6 +1813,36 @@ public void testFastDDLQuery() throws InterruptedException {
     }
   }
 
+  @Test
+  public void testFastQuerySlowDDL() throws InterruptedException {
+    String tableName =
```
Comment (epavan123) [inline google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java:1818]: Please also add a mock test showing that the subsequent getQueryResults call gets made.

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java (hunk 1 above)
Comment (stephaniewang526) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java]: done

File: google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java (hunk 2 above)
Comment (stephaniewang526) [reply google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java:1818]: done

**Review by pmakani — APPROVED** [review]

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java [hunk 3]
```diff
@@ -1268,20 +1268,20 @@ private TableResult queryRpc(
...
+    if (results.getJobComplete() && results.getSchema() != null) {
+      schema = Schema.fromPb(results.getSchema());
       if (results.getNumDmlAffectedRows() == null && results.getTotalRows() == null) {
         numRows = 0L;
       } else if (results.getNumDmlAffectedRows() != null) {
         numRows = results.getNumDmlAffectedRows();
       } else {
         numRows = results.getTotalRows().longValue();
       }
+    } else {
```
Comment (epavan123) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java:1280]: Please add a comment here.

// Query is long running (> 10s) and hasn't completed yet, or query completed but didn't return the schema, fallback. Some operations don't return the schema and can be optimized here, but this is left as future work.

**Review by epavan123 — APPROVED** [review]

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java (hunk 3 above)
Comment (stephaniewang526) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java:1280]: done

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java [hunk 4]
```diff
@@ -1268,20 +1268,23 @@ private TableResult queryRpc(
...
+      // Query is long running (> 10s) and hasn't completed yet, or query completed but didn't
+      // return the schema, fallback. Some operations don't return the schema and can be optimized
+      // here, but this is left as future work.
+      JobId jobId = JobId.fromPb(results.getJobReference());
+      Job job = getJob(jobId, options);
+      TableResult tableResult = job.getQueryResults();
```
Comment (shollyman) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java:1286]: nit: simplify as `return job.getQueryResults();` ?

**Review by shollyman — APPROVED** [review]

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java (hunk 4 above)
Comment (stephaniewang526) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java]: done - ty!

**Pull request merged**